math/geometry/distance_test.go

```diff
@@ -0,0 +1,104 @@
+package geometry
+
+import (
+	"testing"
+)
+
+type args struct {
+	p1 EuclideanPoint
+	p2 EuclideanPoint
+}
+
+func TestFindDistanceBetweenTwoPoints(t *testing.T) {
+	tests := []struct {
+		name    string
+		args    args
+		want    float64
+		wantErr bool
+	}{
+		{
+			"(0,0) and (2,-2)",
+			args{
+				EuclideanPoint{0, 0},
+				EuclideanPoint{2, -2},
+			},
+			2.8284271247461903,
+			false,
+		},
+		{
+			"(-20,23) and (-15,68)",
+			args{
+				EuclideanPoint{-20, 23},
+				EuclideanPoint{-15, 68},
+			},
+			45.27692569068709,
+			false,
+		},
+		{
+			"(2,2) and (14,11)",
+			args{
+				EuclideanPoint{2, 2},
+				EuclideanPoint{14, 11},
+			},
+			15,
+			false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := EuclideanDistance(tt.args.p1, tt.args.p2)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("EuclideanDistance() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if got != tt.want {
+				t.Errorf("EuclideanDistance() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func BenchmarkFindDistanceBetweenTwoPoints(b *testing.B) {
+	benchmarks := []struct {
+		name    string
+		args    args
+		want    float64
+		wantErr bool
+	}{
+		{
+			"(0,0) and (2,-2)",
+			args{
+				EuclideanPoint{0, 0},
+				EuclideanPoint{2, -2},
+			},
+			2.8284271247461903,
+			false,
+		},
+		{
+			"(-20,23) and (-15,68)",
+			args{
+				EuclideanPoint{-20, 23},
+				EuclideanPoint{-15, 68},
+			},
+			45.27692569068709,
+			false,
+		},
+		{
+			"(2,2) and (14,11)",
+			args{
+				EuclideanPoint{2, 2},
+				EuclideanPoint{14, 11},
+			},
+			15,
+			false,
+		},
+	}
```

Comment:
You don't need all these test cases. Pick one and run the benchmark loop on that case only. The benchmark is giving an answer to the question "how long does it take for this algorithm to give answer to three test cases" not "how long does it take to run on one case".

Reply:
Noted and done.